internal/restic/backend.go

```diff
@@ -55,3 +55,23 @@ type FileInfo struct {
 	Size int64
 	Name string
 }
+
+// LoaderBackend TBD
+type LoaderBackend interface {
+	OpenReader(ctx context.Context, h Handle, length int, offset int64) (io.ReadCloser, error)
+}
+
+// DefaultLoaderBackend TBD
+type DefaultLoaderBackend struct {
+	LoaderBackend
+}
+
+// Load implements Backend using lower-level OpenReader func
+func (be *DefaultLoaderBackend) Load(ctx context.Context, h Handle, length int, offset int64, consumer func(rd io.Reader) error) error {
+	rd, err := be.OpenReader(ctx, h, length, offset)
+	if err != nil {
+		return err
+	}
+	defer rd.Close() // ignore errors closing the reader
```

Comment:
Remove the defer, I've been bitten by defer before, and I prefer to have this dealt with explicitly.